server/src/models/room.model.ts

```diff
@@ -0,0 +1,24 @@
+import mongoose, { Schema, Types } from 'mongoose';
+
+interface IRoomModel {
+	name: string;
+	users: Types.ObjectId[];
+	created_at: Date;
+	updated_at: Date;
+}
+
+const roomSchema = new Schema<IRoomModel>(
+	{
+		name: {
+			type: String,
+			unique: true,
+			required: true,
+		},
+		users: [Schema.Types.ObjectId],
+	},
+	{
+		timestamps: true,
```

Comment:
please add  -
group_img - URL of the group profile image; optional for private chats
isGroupChat - Determines if the room is a group or private chat
lastMessage - Link to the most recent message in this room for quick access
with ref: 'Message'

for users better to have ref -
 [{type: mongoose.Schema.Types.ObjectId,
 ref: 'User', // List of users in the room, useful for both groups and private chats}]